app/libs/installations/crashlytics/api.rb

```diff
@@ -0,0 +1,193 @@
+      end
+    end
+
+    # Build the dataset pattern for query purposes
+    def dataset_pattern(dataset)
+      "#{dataset.project_id}.#{dataset.dataset_id}.*"
+    end
+
+    # Firebase APIs
+    def firebase_management_service
+      Installations::Google::Firebase::Api.new(project_number, json_key)
+    end
+
+    def firebase_client
+      execute { firebase_management_service.set_firebase_client }
+    end
+
+    # Query string for Crashlytics data
+    def crashlytics_query(dataset_name)
+      execute { build_crashlytics_query(dataset_name) }
+    end
+
+    # Query string for Analytics data
+    def analytics_query(dataset_name, app_id)
+      execute { build_analytics_query(dataset_name, app_id) }
+    end
+
+    # Run the BigQuery query and fetch the data
+    def get_data(query)
+      bigquery_client.query(query)
+    end
+
+    # Fetch release data (memoized to avoid unnecessary calls)
+    def release_data(app_id)
+      @release_data ||= begin
+        analytics_data = get_data(analytics_query(datasets[:ga4], app_id))
+        crashlytics_data = get_data(crashlytics_query(datasets[:crashlytics]))
+        {analytics_data: analytics_data, crashlytics_data: crashlytics_data}
+      end
+    end
+
+    # Build the SQL query for Crashlytics data
+    def build_crashlytics_query(dataset_name)
+      <<-SQL.squish
+        WITH combined_events AS (
+          SELECT * FROM `#{dataset_name}`
+        ),
+        latest_version AS (
+          SELECT MAX(application.display_version) AS latest_build_version
+          FROM combined_events
+        ),
+        errors_with_version AS (
+          SELECT
+            issue_id,
+            application.display_version AS version_name,
+            application.build_version AS version_code,
+            error_type
+          FROM combined_events
+          WHERE error_type IS NOT NULL
+        ),
+        previous_errors AS (
+          SELECT DISTINCT issue_id
+          FROM errors_with_version
+          WHERE version_name < (SELECT MAX(application.display_version) FROM combined_events)
+        ),
+        new_errors AS (
+          SELECT
+            e.issue_id,
+            e.version_name,
+            e.version_code
+          FROM errors_with_version e
+          JOIN latest_version lv ON e.version_name = lv.latest_build_version
+          LEFT JOIN previous_errors pe ON e.issue_id = pe.issue_id
+          WHERE pe.issue_id IS NULL
+        )
+        SELECT
+          e.version_name AS version_name,
+          COUNT(*) AS errors_count,
+          COUNT(DISTINCT ne.issue_id) AS new_errors_count
+        FROM errors_with_version e
+        LEFT JOIN new_errors ne ON e.issue_id = ne.issue_id
+        GROUP BY e.version_name
+        ORDER BY e.version_name;
+      SQL
+    end
+
+    # Build the SQL query for Analytics data
+    def build_analytics_query(dataset_name, app_id)
+      <<-SQL.squish
+        WITH combined_events AS (
+          SELECT * FROM `#{dataset_name}`
```

Comment:
same suggestion as the other query for limiting the data being processed. we should move the filter `WHERE app_info.firebase_app_id = '#{app_id}'` up from below and add a timestamp limitation of last 15 days.